src/main.rs

```diff
@@ -647,116 +632,146 @@ fn main() {
+                            info!("Restarting rp2040");
+                            if !run_pin.is_set_high() {
+                                run_pin.set_high();
+                                sleep(Duration::from_millis(1000));
+                            }
 
-                                        run_pin.set_low();
-                                        sleep(Duration::from_millis(1000));
-                                        run_pin.set_high();
-                                        break;
-                                    }
-                                    if !frame_acquire{
-                                        break;
+                            run_pin.set_low();
+                            sleep(Duration::from_millis(1000));
+                            run_pin.set_high();
+
+                            if !frame_acquire{
+                                for i in 0..sockets.len(){
+                                    if sockets[i].1.is_some(){
+                                        info!("Shutting down socket {}", if config.use_wifi {"tc2-frames server"} else {sockets[i].0});
+                                        let stream = sockets[i].1.as_mut().unwrap();
+                                        let _ = stream.shutdown().is_ok();
+                                        sockets[i].1 = None;
                                     }
                                 }
+                                break;
+                            }
+                        }
 
-                                let recv_timeout_ms = 10;
-                                let result = rx.recv_timeout(Duration::from_millis(recv_timeout_ms));
-                                match result {
-                                    Ok((Some((radiometry_enabled, is_recording, firmware_version, camera_serial)), None)) => {
-                                        if !config.use_wifi && !sent_header {
-                                            // Send the header info here:
-                                            let model = if radiometry_enabled { "lepton3.5" } else { "lepton3" };
-                                            let header = format!("ResX: 160\nResX: 160\nResY: 120\nFrameSize: 39040\nModel: {}\nBrand: flir\nFPS: 9\nFirmware: DOC-AI-v0.{}\nCameraSerial: {}\n\n", model, firmware_version, camera_serial);
-                                            if let Err(_) = stream.write_all(header.as_bytes()) {
-                                                warn!("Failed sending header info");
-                                            }
+                        let recv_timeout_ms = 10;
+                        let result = rx.recv_timeout(Duration::from_millis(recv_timeout_ms));
+                        match result {
+                            Ok((Some((radiometry_enabled, is_recording, firmware_version, camera_serial)), None)) => {
+                                if !config.use_wifi  {
```

Comment:
It seems now that you've broken the debug wifi case, since this loop doesn't take that into account – again, something stored in the socket tuple about that would probably be clearest.